src/Controller/Command/DeleteLink.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * This File is part of JTL-Software
+ *
+ * User: mbrandt
+ * Date: 24/08/18
+ */
+
+namespace JTL\Onetimelink\Controller\Command;
+
+
+use JTL\Onetimelink\Factory;
+use JTL\Onetimelink\Response;
+use JTL\Onetimelink\Storage\DatabaseStorage;
+use JTL\Onetimelink\User;
+
+class DeleteLink
+{
+    /** @var \JTL\Onetimelink\View\JsonView */
+    private $view;
+
+    /** @var \Monolog\Logger */
+    private $logger;
+
+    /** @var string */
+    private $linkHash;
+
+    /** @var string */
+    private $auth;
+
+    /** @var DatabaseStorage */
+    private $storage;
+
+    /** @var Factory */
+    private $factory;
+
+    public function __construct(
+        DatabaseStorage $storage,
+        Factory $factory,
+        string $linkHash,
+        string $auth
+    )
+    {
+        $this->view = $factory->createJsonView();
+        $this->logger = $factory->createLogger();
+        $this->linkHash = $linkHash;
+        $this->auth = $auth;
+        $this->storage = $storage;
+        $this->factory = $factory;
+    }
+
+    public function execute(): Response
+    {
+        $this->logger->info('Requested delete link auth for link hash ' . $this->linkHash . ' with auth '
+                            . $this->auth);
+        $ownerEmail = $this->storage->getDeleteAuthOwnerEmail($this->linkHash, $this->auth);
```

Comment:
Use Username instead of Email for Naming. Because it is our OTL Username. Right?

Reply:
What username? There are no usernames in OTL

Reply:
Mhm ok. :(

I dont't like the wording "email" in this context. But ok I can live with that... for now. ;)